test/irmin-pack/test_indexing_strategy.ml

```diff
@@ -0,0 +1,131 @@
+(*
+ * Copyright (c) 2023 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open! Import
+open Common
+
+let src =
+  Logs.Src.create "tests.indexing_strategy" ~doc:"Test indexing strategy"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Store = struct
+  module Maker = Irmin_pack_unix.Maker (Conf)
+  include Maker.Make (Schema)
+end
+
+let test_dir =
+  let ( / ) = Filename.concat in
+  "_build" / "test-indexing_strategy"
+
+let () = Irmin_pack_unix.Io_legacy.Unix.mkdir test_dir
```

Comment:
nitpick, if you use `test_dir` as root, then opening the store with `fresh:true` will create the dir

Reply:
Hmm, the first open is `fresh:true` but it gave an error. I think because the intermediate directory didn't exist. Is it expected that `fresh:true` will create all directories in the path?

(side comment: I just noticed the mixture of `-` and `_` in the directory name. :see_no_evil: I'll change that)

Reply:
Oh, ha, I just understood your suggestion to use the test directory _as_ the root. :facepalm: That is simpler. I'll make this change.